src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            return session.createQuery("SELECT r FROM Role r WHERE "
+                            + "r.roleName =: roleName", Role.class)
+                    .setParameter("roleName", Role.RoleName.valueOf(roleName))
```

Comment:
```suggestion
            return session.createQuery("FROM Role r WHERE "
                            + "r.roleName =: roleName", Role.class)
                    .setParameter("roleName", roleName)
```

Reply:
without Role.Role Name.valueOf(role Name)) we will get an Exception